test/tests/hostmask.js

```diff
@@ -0,0 +1,25 @@
+"use strict";
+
+const expect = require("chai").expect;
+const Helper = require("../../src/helper");
+
+describe("Hostmask", function() {
+	it("should correctly split hostmask components", function() {
+		expect(Helper.parseHostmask("nick!user@host").nick).to.equal("nick");
+		expect(Helper.parseHostmask("nick!user@host").ident).to.equal("user");
+		expect(Helper.parseHostmask("nick!user@host").hostname).to.equal("host");
+	});
+
+	it("fallback to wildcard if components are empty", function() {
```

Comment:
I would like to see more tests for this trying to cover odd cases like empty strings, `!@`, `nick!!!!@thing@@host` and some others.

`compareHostmask` should have some more tests too, where nick is wildcard but hostname is not, etc.